### PR TITLE
feat: implemented filtering users by coding language

### PR DESCRIPTION
**Pull request by Ekep-Obasi**
### Describe your changes

Implemented filter for people by skills (coding language) on the people page

### Preview

https://www.loom.com/share/abe53334cf43409e88213ecb188faaa5?sid=c1f9a146-83b9-46ee-a49f-c28ed59278b0

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
Good work. LGTM.

File: frontend/app/src/people/main/Body.tsx [hunk 1]
```diff
@@ -85,6 +100,29 @@ function BodyComponent() {
...
+
+    return users.filter((user: PersonType) => {
+      const userCodingLanguages = (user.extras.coding_languages ?? []).map(
+        (t: { [key: string]: string }) => t.value
+      );
+      return requiredLanguages?.every((requiredLanguage: string) =>
+        userCodingLanguages.includes(requiredLanguage)
+      );
+    });
+  };
```
Comment (kevkevinpal) [inline frontend/app/src/people/main/Body.tsx:120]: we should put this in a util and write a test for it

**Comment by kevkevinpal** [comment]
This PR looks good to me we should just move that one function into a util and write a test for it and then we should be good to merge

File: frontend/app/src/people/main/Body.tsx (hunk 1 above)
Comment (Ekep-Obasi) [reply frontend/app/src/people/main/Body.tsx:120]: Okay @kevkevinpal

**Comment by Ekep-Obasi** [comment]
@ecurrencyhodler 
I ran prettier:checks before pushing, i don't know why it kept failing

**Comment by ecurrencyhodler** [comment]
Hmmm @Ekep-Obasi something must've gotten wrong in staging. Can you take another look? Here's a screenshot. I don't see the filtering options for some reason on the top and the # of profiles displayed looks like they've been pre-filtered.

https://people-test.sphinx.chat/p
Screenshot 2023-12-08 at 10 23 55 PM

**Comment by Ekep-Obasi** [comment]
@ecurrencyhodler,  please check that the deployment is up to date 
The changes show locally on the master branch it should be okay!

**Comment by ecurrencyhodler** [comment]
Okay I will check.

**Comment by ecurrencyhodler** [comment]
Alright I see the filter now but I still only see a few profiles. Do you know why that might be?
Screenshot 2023-12-09 at 8 24 23 AM

**Comment by Ekep-Obasi** [comment]
Let me take a look

**Comment by ecurrencyhodler** [comment]
I'll ask talk with the team in our standup Monday.

**Comment by ecurrencyhodler** [comment]
Hey @Ekep-Obasi sorry for the delay. I was unable to test due to the issue of not seeing a lot of profiles on the people page. We have since submitted a fix and I have tested it and it works! Great job. Paying you out now.

**Pull request merged**